state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -474,12 +478,28 @@ impl L1Fetcher {
                             contracts.v2.functions_by_name("commitBatches").unwrap()[0].clone();
                     }
 
-                    let blocks = match parse_calldata(block_number, &function, &tx.input).await {
-                        Ok(blks) => blks,
-                        Err(e) => {
-                            tracing::error!("Failed to parse calldata: {e}");
-                            cancellation_token.cancel();
-                            return last_block_number_processed;
+                    let blocks = loop {
```

Comment:
We should not handle blob errors here, that should only be done within `V3`.

Reply:
That isn't possible. On the one hand the program must retry on network errors, but on the other hand it cannot retry forever to remain cancellable - so there are two places where retrieval is retried: the high level has access to the `CancellationToken` while the low level has a hard-coded retry limit. I admit it's inelegant, but refactoring the whole fetcher around a network retrieval loop would IMHO be too drastic change... Note that `retry_call` works the same way.

Reply:
Yes, ideally, but I'm afraid getting that through the layers of the ethers provider would be non-trivial...